### DISCUSSION TITLE
Schedule checking

### DISCUSSION THREAD

**Discussion by alpharde** [moosefs/moosefs#605] [Q&A]
I would like to know if there's a way to schedule or trigger file checking because I'm using it for "hot archival" and from what I've been told or read in the issues, it seems that MooseFS will (only?) check files for consistency during read time.

**Comment by tianon** [comment]
The good news is that last bit isn't accurate :smile:

One of the best features (IMO) of MooseFS is that it's _constantly_ re-checking the consistency of chunks at rest (and then proactively making new copies if it detects a problem):

https://github.com/moosefs/moosefs/blob/5afd7b57c065d6b36ebe755ccffd17adf33b1975/mfsmanpages/mfsmaster.cfg.5#L206-L210

(I'm sure there's a better reference for this functionality, but this is the first one that comes to mind for me)

**Reply by alpharde** [reply]
Ah, thanks. I was mistaken then.

**Comment by chogata** [comment]
@tianon is right that MooseFS is constantly re-checking the consistency of chunks at rest, but the config options quoted are for something else - for chunk loop in the main master, that is tasked with replicating chunks that are undergoal, on wrong labels or just to rebalance disk usage on all machines.
It's the chunkservers that check the consistency of the chunk copies they keep, they do it continuously in one long loop and this loop is governed by these 2 variables in `mfchunkserver.cfg` file:

```
# speed of background chunk tests in MB/s per disk (formally entry defined in mfshdd.cfg). Value can be given as a decimal number (default is 1.0)
# HDD_TEST_SPEED = 1.0

# do not test a chunk's integrity when last I/O (including test) on this chunk was performed less than HDD_MIN_TEST_INTERVAL ago (default is 1 day)
# HDD_MIN_TEST_INTERVAL = 1d
```

Basically, the chunkserver will continuously  read all the chunk copies it has stored on disks, compare the data content to checksums in the header and if anything is wrong (i/o error on disk, checksum not matching the data), this copy will be reported to the master and flagged as invalid. Then, the whole chunk goes into undergoal status and the master finds another chunk server with a good copy and yet another that can host a copy and tells them to perform a replication operation.

For more information on how to adjust the configuration if necessary (e.g. how to define time intervals) refer to manpage https://moosefs.com/manpages/mfschunkserver-cfg
